setup.py

```diff
@@ -14,31 +14,30 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 """
-import sys
-from setuptools import setup, find_packages
 
-install_requires = []
-if sys.platform.startswith('linux'):
-    install_requires.extend([
-        'pyusb',
-    ])
-elif sys.platform.startswith('win'):
-    install_requires.extend([
-        'pywinusb',
-    ])
-elif sys.platform.startswith('darwin'):
-    install_requires.extend([
-        'hidapi',
-    ])
+from setuptools import setup, find_packages
+import sys
 
+install_requires = dict(
+    win32=['pyWinUSB'],
+    linux2=['pyUSB'],
+    darwin=['hidapi']
+)
 
 setup(
     name="pyOCD",
-    version="0.4.2",
+    use_scm_version={
+        'local_scheme': 'dirty-tag',
+        'write_to': 'pyOCD/_version.py'
+    },
+    setup_requires=['setuptools_scm'],
     description="CMSIS-DAP debugger for Python",
+    long_description=open('README.md', 'Ur').read(),
```

Comment:
`long_description` actually needs to be restructured text.  I have done something like this on other projects in order to work around this: https://github.com/digidotcom/python-wvalib/blob/master/setup.py#L27